### DISCUSSION TITLE
Realtime API

### DISCUSSION THREAD

**Discussion by riekusr** [allanbunch/node-red-openai-api#25] [Ideas]
Hi, is streaming API going to be supported? 
These new voice realtime API will be great to create voice agents (in combination with twilio)
https://www.youtube.com/watch?v=AZ0WziqO_QA

**Comment by allanbunch** [comment]
@riekusr Absolutely. Excited to get the Realtime API implemented. I'm going to spend some time on this toward the end of the week. Stay tuned for an update.

**Reply by riekusr** [reply]
Awesome, can't wait!

**Reply by allanbunch** [reply]
I spent some time getting familiar with the Realtime API and I'm absolutely impressed. I'm torn as to whether it makes sense to drop a Realtime API method into my OpenAI API node. OpenAI's implementation with this is very different from their standard API--websockets, etc. And I found the flexibility of using standard Node-RED nodes for this very intriguing. Got me thinking about lots of innovative things we can do with this in Node-RED. Stay tuned while I give this some thought.

In the meantime, check out the [demo Node-RED flow](https://flows.nodered.org/flow/cb0123996b8ccbc8f026ebe780bfd76d) I created, putting the Realtime API into action inside of Node-RED! This version's a base implementation, though I did stub out a flow path for function call responses. Give the flow a run and let me know how it goes.
